core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -777,6 +787,75 @@ NavigableMap<Long, InFlightBatch> cachedState() {
         return new ConcurrentSkipListMap<>(cachedState);
     }
 
+    // TODO: maxDeliveryCount should be utilized here once it is implemented
+    /**
+     * Apply acquisition lock to acquired records.
+     * @param memberId The member id of the client that is putting the acquisition lock.
+     * @param baseOffset The base offset of the acquired records.
+     * @param lastOffset The last offset of the acquired records.
+     * @param cachedStateEntry The cached state map entry for the acquired records which contains the entire batch
+     *                         from base offset to last offset.
+     */
+    private void startAcquisitionLockTimer(String memberId, long baseOffset, long lastOffset, Map.Entry<Long, InFlightBatch> cachedStateEntry) {
+        timer.add(new TimerTask(recordLockDurationMs) {
+            // Runs when the acquisition lock timer expires. We would then be releasing the acquired records.
+            @Override
+            public void run() {
+                lock.writeLock().lock();
+                try {
+                    InFlightBatch inFlightBatch = cachedStateEntry.getValue();
+                    long localNextFetchOffset = nextFetchOffset;
+                    // Case when the state of complete batch is valid
+                    if (inFlightBatch.offsetState == null) {
+                        if (inFlightBatch.batchState() == RecordState.ACQUIRED) {
+                            InFlightState updateResult = inFlightBatch.tryUpdateBatchState(RecordState.AVAILABLE, false);
+                            if (updateResult == null) {
+                                log.debug("Unable to release acquisition lock on timeout for the batch: {}"
+                                        + " for the share partition: {}-{}-{}", inFlightBatch, groupId, memberId, topicIdPartition);
+                            } else {
+                                localNextFetchOffset = Math.min(cachedStateEntry.getKey(), localNextFetchOffset);
+                            }
+                        } else {
+                            log.trace("The batch is not in acquired state while release of acquisition lock on timeout, skipping, batch: {}"
```

Comment:
Since, we were not cancelling timer tasks, this situation can occur if records are acknowledged. Agreed that this situation should have warn/debug/error in case we also cancel tasks.

Reply:
Changed them to DEBUG  since the functionality of cancelling tasks is available now.